### PR TITLE
Develop contribution documentation

### PR DESCRIPTION
**Pull request by linglp**
Add information related to poetry in our contribution guide.

### CODE REVIEW

**Comment by mialy-defelice** [comment]
Change `pip install` to `poetry add`

**Comment by linglp** [comment]
if you are using token-based authentication, you would need to change the test_config.yml. Tests are failing because this is from a forked branch.

File: tests/data/test_config.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 definitions:
   creds_path: "../../credentials.json"
   token_pickle: "token.pickle"
+  synapse_config: "../../.synapseConfig"
```
Comment (milen-sage) [inline tests/data/test_config.yml:4]: @linglp could you add a comment here that this is needed only if you have used the Synapse token authentication approach?

**Comment by milen-sage** [comment]
@linglp #604 handles some of this documentation but not other? Would it make sense to first merge 604 and then open a new PR with any changes here that are outside of #604?

**Comment by linglp** [comment]
@milen-sage most of the changes in contribution.md have already been included in other PR. But we will still need to the update the documentation for API and add comments for test_config.yml. I will open a separate PR for these and close this one for now.

**Pull request closed without merging**